### PR TITLE
Add generalized scripts to run generated payloads

### PR DESCRIPTION
**Pull request by alejoberardino**
## Description

This PR adds scripts to run the single command docker environment and run generated payloads to reproduce NOP environments with internal tooling.

## Quality Assurance

- [ ] Ran `yarn changeset` if adapter source code was changed
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [ ] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [ ] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [ ] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/k6/src/repro-test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { check, sleep } from 'k6'
+import { SharedArray } from 'k6/data'
+import http, { BatchRequests } from 'k6/http'
+import { Rate } from 'k6/metrics'
+import { Payload } from './config/types'
+
+// load the test duration from the environment or default to 12 hours
```
Comment (austinborn) [inline packages/k6/src/repro-test.ts:7]:
```suggestion
// load the test iterations from the environment or default to 100
```

File: packages/k6/src/repro-test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      'Content-Type': 'application/json',
+    },
+  }
+
+  const adapterName = __ENV.LOAD_TEST_ADAPTER_NAME
+  const adapterUrl = `http://${adapterName}-adapter:8080`
+
+  for (const payload of payloadData) {
+    console.log('data:', payload.data)
+    console.log('parsed data', JSON.parse(payload.data))
```
Comment (austinborn) [inline packages/k6/src/repro-test.ts:48]: Are these necessary?

File: packages/k6/src/repro-test.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  const responses = http.batch(batchRequests)
+  for (const [name, response] of Object.entries(responses)) {
+    const result = check(response, {
+      [`${name} returned 200`]: (r) => r.status == 200,
+    })
+
+    errorRate.add(!result)
+  }
+
+  sleep(1)
```
Comment (austinborn) [inline packages/k6/src/repro-test.ts:74]: I know this was copied from another file, but do you know why we `sleep(1)` here?

File: packages/k6/src/repro-test.ts (hunk 2 above)
Comment (alejoberardino) [reply packages/k6/src/repro-test.ts:48]: No, will remove 👍

File: packages/k6/src/repro-test.ts (hunk 3 above)
Comment (alejoberardino) [reply packages/k6/src/repro-test.ts]: Not really haha, I removed it and everything seems to work fine

File: grafana/scripts/compose.sh [hunk 4]
```diff
@@ -11,4 +11,7 @@ function join_by {
 CONTEXT='..' yarn generate:docker-compose
 ./scripts/generate-prom-config.sh "$(join_by , "${arr[@]}")"
 
-docker-compose -f ./docker-compose.yaml -f ../docker-compose.generated.yaml up --build redis grafana prometheus "${arr[@]}"
+docker-compose \
+-f ./docker-compose.yaml \
+-f ../docker-compose.generated.yaml \
+up $DETACHED redis grafana prometheus "${arr[@]}" $LOAD_TEST_SERVICE
```
Comment (austinborn) [inline grafana/scripts/compose.sh:17]: Does `DETACHED` have to be set to `build`? How does the command work now?

File: packages/k6/src/repro-test.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  },
+}
+
+export const errorRate = new Rate('errors')
+
+// load the test data, if data was generated then load it from the generated file
+if (!__ENV.PAYLOAD_PATH) {
+  console.error('PAYLOAD_PATH env var is not present')
+  process.exit(1)
+}
```
Comment (austinborn) [inline packages/k6/src/repro-test.ts:29]: Is it not safe to default `PAYLOAD_PATH` to `'../src/config/http.json'`?

**Review by austinborn — COMMENTED** [review]
Looks good code-wise, but could you add documentation on when to use these tests and requirements to run it? An explanation in `packages/k6/README.md` would be great.

File: packages/k6/src/repro-test.ts (hunk 5 above)
Comment (alejoberardino) [reply packages/k6/src/repro-test.ts:29]: For now this is separate from the other testing workflow, in the future this file could also serve to perform the same tests as the other

File: package.json [hunk 6]
```diff
@@ -27,6 +27,7 @@
     "generate:image-name": "ts-node-transpile-only ./packages/scripts/src/generate-image-name",
     "generate:master-list": "ts-node-transpile-only ./packages/scripts/src/generate-master-list",
     "generate:readme": "ts-node-transpile-only ./packages/scripts/src/generate-readme",
+    "repro": "./packages/scripts/src/repro/run.sh",
```
Comment (austinborn) [inline package.json:30]: Does this command work?
```
~/chainlink/external-adapters-js [rdd] $ yarn repro         
command not found: ./packages/scripts/src/repro/run.sh
```

Comment (austinborn) [inline package.json:30]: Does this command work?
```
~/chainlink/external-adapters-js [rdd] $ yarn repro         
command not found: ./packages/scripts/src/repro/run.sh
```

File: packages/k6/src/repro-test.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
+import { check, sleep } from 'k6'
```
Comment (austinborn) [inline packages/k6/src/repro-test.ts:1]:
```suggestion
import { check } from 'k6'
```

File: grafana/scripts/compose.sh (hunk 4 above)
Comment (austinborn) [reply grafana/scripts/compose.sh:17]: got it 👍

Comment (austinborn) [reply grafana/scripts/compose.sh:17]: got it 👍

**Pull request merged**